caravel/models.py

```diff
@@ -154,42 +154,37 @@ class Slice(Model, AuditMixinNullable):
     __tablename__ = 'slices'
     id = Column(Integer, primary_key=True)
     slice_name = Column(String(250))
-    druid_datasource_id = Column(Integer, ForeignKey('datasources.id'))
-    table_id = Column(Integer, ForeignKey('tables.id'))
+    datasource_id = Column(Integer)
     datasource_type = Column(String(200))
     datasource_name = Column(String(2000))
     viz_type = Column(String(250))
     params = Column(Text)
     description = Column(Text)
     cache_timeout = Column(Integer)
     perm = Column(String(2000))
-
-    table = relationship(
-        'SqlaTable', foreign_keys=[table_id], backref='slices')
-    druid_datasource = relationship(
-        'DruidDatasource', foreign_keys=[druid_datasource_id], backref='slices')
     owners = relationship("User", secondary=slice_user)
 
     def __repr__(self):
         return self.slice_name
 
     @property
+    def cls_model(self):
+        return src_registry.get_cls_model(self.datasource_type)
+
+    @property
```

Comment:
Having a property here hides the fact that there's a db request every time the attribute is accessed. I think that before with the relationship there would be a single database request with a join, now it runs another query every time this property is referenced.

Removing @property and calling the method `get_datasource` will reveal everywhere where it's getting used, that way we can make sure it's not getting accessed many times.